be/src/exprs/string_functions.cpp

```diff
@@ -3501,6 +3501,150 @@ StatusOr<ColumnPtr> StringFunctions::regexp_extract_all(FunctionContext* context
     return regexp_extract_all_general(context, options, columns);
 }
 
+Status StringFunctions::regexp_instr_prepare(FunctionContext* context, FunctionContext::FunctionStateScope scope) {
+    // init RE2 options
+    if (scope != FunctionContext::THREAD_LOCAL) {
+        return Status::OK();
+    }
+
+    auto* state = new StringFunctionsState();
+    context->set_function_state(scope, state);
+
+    state->options = std::make_unique<re2::RE2::Options>();
+    state->options->set_log_errors(false);
+    state->options->set_longest_match(false);
+    state->options->set_dot_nl(true);
+
+    // go row regex
+    if (!context->is_notnull_constant_column(1)) {
+        return Status::OK();
+    }
+
+    state->const_pattern = true;
+    auto column = context->get_constant_column(1);
+    auto pattern = ColumnHelper::get_const_value<TYPE_VARCHAR>(column);
+    state->pattern = pattern.to_string();
+    state->regex = std::make_unique<re2::RE2>(state->pattern, *(state->options));
+
+    if (!state->regex->ok()) {
+        std::stringstream error;
+        error << "Invalid regex expression: " << pattern.to_string();
+        context->set_error(error.str().c_str());
+        return Status::InvalidArgument(error.str());
+    }
+
+    return Status::OK();
+}
+
+static ColumnPtr regexp_instr_const(StringFunctionsState* state, const Columns& columns) {
+    auto content_viewer = ColumnViewer<TYPE_VARCHAR>(columns[0]);
+    re2::RE2* const_re = state->get_or_prepare_regex();
+    std::string pattern_str = state->pattern;
+
+    auto size = columns[0]->size();
+    ColumnBuilder<TYPE_INT> result(size);
+    for (int row = 0; row < size; ++row) {
+        if (pattern_str.empty()) {
+            result.append(1);
+            continue;
+        }
+
+        if (content_viewer.is_null(row)) {
+            result.append(0);
+            continue;
+        }
+
+        auto row_value = content_viewer.value(row);
+        
+        re2::StringPiece str_sp(row_value.get_data(), row_value.get_size());
+        int max_matches = 1 + const_re->NumberOfCapturingGroups();
+        std::vector<re2::StringPiece> matches(max_matches);
+        bool success = const_re->Match(str_sp, 0, row_value.get_size(), re2::RE2::UNANCHORED, &matches[0], max_matches);
+        if (!success) {
+            result.append(0);
+            continue;
+        }
+
+        const re2::StringPiece& match = matches[0];
+        std::string row_str = row_value.get_data();
+        std::string match_str = match.data();
+
+        auto index = row_str.find(match_str);
```

Comment:
suggest to use instr function instead of find, the former outperforms the the latter.